### PR TITLE
fix(changesets): add missing `pnpm` command, pre-build project

### PR DESCRIPTION
**Pull request by szilarddoro**

### CODE REVIEW

**Comment by szilarddoro** [comment]
What do you about this change in the publish step @plmercereau? I was thinking, that maybe it's better if we create a separate "Publish to Vercel" step that simply depends on the Docker publish.

**Comment by plmercereau** [comment]
why would we want to add this step? it is not needed in the next steps, and as it is already executed in a dependent job the turborepo remote cache should already be available when building the images.

Yes, why not creating a separate job for Vercel

**Comment by plmercereau** [comment]
> why would we want to add this step? it is not needed in the next steps, and as it is already executed in a dependent job the turborepo remote cache should already be available when building the images.
> 
> Yes, why not creating a separate job for Vercel

Oh, i missed the fact the build+deploy to vercel was happening inside the GH action - thought we were still triggering the hook.
Nice

File: .github/workflows/changesets.yaml [hunk 1]
```diff
@@ -129,3 +122,26 @@ jobs:
...
+        with:
+          TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
+          TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
+      - name: Setup Vercel CLI
+        run: pnpm add -g vercel
+      - name: Trigger a Vercel deployment
+        env:
+          VERCEL_ORG_ID: ${{ secrets.DASHBOARD_VERCEL_TEAM_ID }}
+          VERCEL_PROJECT_ID: ${{ secrets.DASHBOARD_VERCEL_PROJECT_ID }}
+        run: vercel pull --environment=production && vercel build --prod && vercel deploy --prebuilt --prod
```
Comment (plmercereau) [inline .github/workflows/changesets.yaml:147]: Maybe multiple lines instead of multiple `&&` to improve readability
GH actions have `set -e` anyway

**Review by plmercereau — APPROVED** [review]

File: .github/workflows/changesets.yaml (hunk 1 above)
Comment (szilarddoro) [reply .github/workflows/changesets.yaml:147]: > > why would we want to add this step? it is not needed in the next steps, and as it is already executed in a dependent job the turborepo remote cache should already be available when building the images.
> > Yes, why not creating a separate job for Vercel
> 
> Oh, i missed the fact the build+deploy to vercel was happening inside the GH action - thought we were still triggering the hook. Nice

Nope. Vercel runs the build step ignore script even if we trigger the deployment through the webhook. We ignored `main` branch on purpose for our production deployment, but that meant that even webhooks would be ignored, so we need to use the CLI.

**Comment by szilarddoro** [comment]
> Oh, i missed the fact the build+deploy to vercel was happening inside the GH action - thought we were still triggering the hook. Nice

Nope. Vercel runs the build step ignore script even if we trigger the deployment through the webhook. We ignored `main` branch on purpose for our production deployment, but that meant that even webhooks would be ignored, so we need to use the CLI.

**Pull request merged**